lib/mainloop.c

```diff
@@ -336,6 +340,35 @@ main_loop_get_current_config(MainLoop *self)
   return self->current_configuration;
 }
 
+/* main_loop_verify_config(GString *result, MainLoop *self)
+ * compare current comfiguration against config file on disk */
+
+void
+main_loop_verify_config(GString *result, MainLoop *self)
+{
+  const gchar *str;
+  const gchar *config_path = resolvedConfigurablePaths.cfgfilename;
+  gchar *config_mem =  self -> current_configuration -> original_config -> str;
+  gchar *config_file;
+  
+  gchar *buff;
+  GString *content = g_string_new("");
+  if (g_file_get_contents(config_path, &buff, NULL, NULL))
```

Comment:
I would report a different warning if the config file does not exists, that could be helpful and easy to do here.

(I know I wrote a code like this, but at that place it is guaranteed that the same file will be opened, so access issue will be reported.)